store/gaskvstore.go

```diff
@@ -69,7 +62,18 @@ func (gi *gasKVStore) Delete(key []byte) {
 
 // Implements KVStore
 func (gi *gasKVStore) Prefix(prefix []byte) KVStore {
-	return prefixStore{gi, prefix}
+	// Keep gasstore layer at the top
+	return &gasKVStore{
+		gasMeter:  gi.gasMeter,
+		gasConfig: gi.gasConfig,
+		parent:    prefixStore{gi.parent, prefix},
+	}
+}
+
+// Implements KVStore
+func (gi *gasKVStore) Gas(meter GasMeter, config GasConfig) KVStore {
+	// Override existing meter/config
```

Comment:
Hmm - a bit counter-intuitive, what if we wanted to track gas on two different meters for some reason? I guess this is OK for now though.

Reply:
That makes sense, I will change it not to override existing settings.